provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -406,13 +402,16 @@ func (c *Client) selectDatastore(
 	if err := c.client.Retrieve(ctx, refs, nil, &datastores); err != nil {
 		return nil, errors.Annotate(err, "retrieving datastore details")
 	}
-	if args.Datastore != "" {
+	if args.Constraints.RootDiskSource != nil {
+		dsName := *args.Constraints.RootDiskSource
+		c.logger.Debugf("desired datasource %q", dsName)
 		for _, ds := range datastores {
-			if ds.Name == args.Datastore {
+			c.logger.Debugf("seen %q", ds.Name)
+			if ds.Name == dsName {
 				return &ds, nil
 			}
 		}
-		return nil, errors.Errorf("could not find datastore %q", args.Datastore)
+		return nil, errors.Errorf("could not find datastore %q", dsName)
```

Comment:
I think there's a pre-checks bit in the workflow - could be use that to fail early? Would mean that juju deploy exists with an error

Reply:
Good call - looking into that now.